src/components/implementation/cbuf_mgr/default/cbuf_mgr.c

```diff
@@ -755,18 +1062,93 @@ cbuf_register(spdid_t spdid, long cbid)
 
 	printl("cbuf_register\n");
 	CBUF_TAKE();
+	tracking_start(NULL, CBUF_REG);
+
 	cci = cbuf_comp_info_get(spdid);
-	if (!cci) goto done;
+	if (unlikely(!cci)) goto done;
 	cmr = cbuf_meta_lookup_cmr(cci, cbid);
 	if (cmr) ERR_THROW(cmr->dest, done);
 
 	/* Create the mapping into the client */
-	if (cbuf_alloc_map(spdid, &dest, &p, PAGE_SIZE)) goto done;
-	assert((u32_t)p == round_to_page(p));
+	if (cbuf_alloc_map(spdid, &dest, &p, NULL, PAGE_SIZE, MAPPING_RW)) goto done;
+	assert((unsigned int)p == round_to_page(p));
 	cmr = cbuf_meta_add(cci, cbid, p, dest);
 	assert(cmr);
 	ret = cmr->dest;
 done:
+	tracking_end(NULL, CBUF_REG);
+
+	CBUF_RELEASE();
+	return ret;
+}
+
+static void
+cbuf_shrink(struct cbuf_comp_info *cci, int diff)
+{
+	int i, sz;
+	struct cbuf_bin *bin;
+	struct cbuf_info *cbi, *next, *head;
+
+	for (i = cci->nbin-1 ; i >= 0 ; i--) {
```

Comment:
PLEASE add blank lines.  Humans communicate in paragraphs, not in walls of text.